PyRoute/TradeCodes.py

```diff
@@ -16,16 +16,54 @@ class TradeCodes(object):
     pcodes = ['As', 'De', 'Ga', 'Fl', 'He', 'Ic', 'Oc', 'Po', 'Va', 'Wa']
     dcodes = ['Cp', 'Cx', 'Cs', 'Mr', 'Da', 'Di', 'Pz', 'An', 'Ab', 'Fo', 'Px',
               'Re', 'Rs', 'Sa', 'Tz', 'Lk',
-              'RsA', 'RsB', 'RsG', 'RsD', 'RsE', 'RsZ', 'RsT', 'RsI', 'RsK',
+              'RsA', 'RsB', 'RsG', 'RsD', 'RsE', 'RsZ', 'RsT', 'RsI', 'RsK', 'RsO',
               'Fr', 'Co', 'Tp', 'Ho', 'Tr', 'Tu',
               'Cm', 'Tw']
     ex_codes = {'As', 'Fl', 'Ic', 'De', 'Na', 'Va', 'Wa', 'He', 'Oc'}
     research = {'RsA': '\u0391', 'RsB': '\u0392', 'RsG': '\u0393',
-                'RsD': '\u0394', 'RdE': '\u0395', 'RsZ': '\u0396',
+                'RsD': '\u0394', 'RdE': '\u0395', 'RsO': '\u03A9', 'RsZ': '\u0396',
                 'RsT': '\u0398', 'RsI': '\u0399', 'RsK': '\u039A'}
     pcolor = {'As': '#8E9397', 'De': '#d17533', 'Fl': '#e37dff', 'He': '#ff6f0c', 'Ic': '#A5F2F3',
               'Oc': '#0094ED', 'Po': '#6a986a', 'Va': '#c9c9c9', 'Wa': '#4abef4'}
     ext_codes = {'Lt', 'Ht', 'Lg', 'Hg'}
+    weird_codes = {'{Anomaly}', '{Fuel}', '{Ringworld}', '{Rosette}'}
+    allowed_residual_codes = {'Ag', 'Ba', 'Bo', 'Cl', 'Cw', 'Cy', 'Dw', 'Ex', 'Fr', 'Ga', 'Hi', 'In', 'Lo', 'N1', 'Na',
+                              'Ni', 'o', 'Pa', 'Ph', 'Pi', 'Po', 'Pr', 'Ri', 'Rn', 'Rv', 's', 'Sp', 'St', 'Tn', 'Za'}
+    # Whether any of these pairs are _permitted_ to occur together is _irrelevant_.  They _do_ occur together in the
+    # TravellerMap raw data.
+    ok_pairs = {
```

Comment:
I understand this list is generated from the whole Traveller Map data set. But this contains pairs which are not valid under the T5 rules, and therefor for the reviewed sectors. At some point there will need to be a "T5_allowed_pairs" which will be much smaller than this to support the review process.